packages/blob/src/helpers.ts

```diff
@@ -32,12 +35,64 @@ export class BlobAccessError extends Error {
   }
 }
 
+export class BlobNotFoundError extends Error {
+  constructor() {
+    super('Vercel Blob: This store does not exist');
+  }
+}
+
+export class BlobSuspendedError extends Error {
+  constructor() {
+    super('Vercel Blob: This store has been suspended');
+  }
+}
+
 export class BlobUnknownError extends Error {
   constructor() {
     super('Vercel Blob: Unknown error, please visit https://vercel.com/help');
   }
 }
 
+type BlobApiErrorCodes =
+  | 'suspended_store'
+  | 'forbidden'
+  | 'not_found'
+  | 'bad_request';
+
+type BlobApiError = { error?: { code: BlobApiErrorCodes } } | undefined;
+
+export async function validateBlobApiResponse(
+  response: Response
+): Promise<Error | undefined> {
+  if (response.status !== 200) {
+    if (response.status < 500) {
+      let data: BlobApiError;
+      try {
+        data = (await response.json()) as BlobApiError;
```

Comment:
I wonder if this is 100% correct: when you use .json() on a response, then you cannot use it again later on.
Example: if we receive 404 for a non-existing blob, then we will try to consume .json in this method, then we will try again in the head method and this will throw.

Even if our code may acknowledge for that I think `validateBlobApiResponse` should either throw OR return the json data to be used in the rest of the program, that would be easier to follow and less error prone.

Thoughts?

Reply:
good point, I will update this code 👍

Reply:
something like:

```
const apiResponse = validateBlobApiResponse
```